src/FakeItEasy/Core/SequentialCallContext.cs

```diff
@@ -2,48 +2,51 @@ namespace FakeItEasy.Core
 {
     using System;
     using System.Collections.Generic;
-    using System.Globalization;
-    using System.IO;
+    using System.Linq;
 
-    internal class OrderedFakeAsserter : IFakeAsserter
+    internal class SequentialCallContext : ISequentialCallContext
     {
-        private readonly IEnumerable<IFakeObjectCall> originalCallList;
         private readonly CallWriter callWriter;
-        private readonly Queue<IFakeObjectCall> calls;
-        private readonly List<AssertedCall> assertedCalls = new List<AssertedCall>();
+        private readonly HashSet<FakeManager> fakeManagers;
+        private readonly List<AssertedCall> assertedCalls;
+        private int currentSequenceNumber;
 
-        public OrderedFakeAsserter(IEnumerable<IFakeObjectCall> calls, CallWriter callWriter)
+        public SequentialCallContext(CallWriter callWriter)
         {
-            Guard.AgainstNull(calls, "calls");
-            Guard.AgainstNull(callWriter, "callWriter");
-
-            this.originalCallList = calls;
-            this.calls = new Queue<IFakeObjectCall>(calls);
             this.callWriter = callWriter;
+            this.currentSequenceNumber = -1;

```

Comment:
:lipstick: if we're touching this class again to restore the guard clause (see above) then I guess we could match the order of these assignments to the field declaration order.